classes/class-wc-connect-payment-methods-store.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+if ( ! class_exists( 'WC_Connect_Payment_Methods_Store' ) ) {
+
+	class WC_Connect_Payment_Methods_Store {
+
+		/**
+		 * @var WC_Connect_Service_Settings_Store
+		 */
+		protected $service_settings_store;
+
+		/**
+		 * @var WC_Connect_API_Client
+		 */
+		protected $api_client;
+
+		/**
+		 * @var WC_Connect_Logger
+		 */
+		protected $logger;
+
+		public function __construct( WC_Connect_Service_Settings_Store $service_settings_store,
+			WC_Connect_API_Client $api_client, WC_Connect_Logger $logger ) {
+
+			$this->service_settings_store = $service_settings_store;
+			$this->api_client             = $api_client;
+			$this->logger                 = $logger;
+
+		}
+
+		public function fetch_payment_methods_from_connect_server() {
+
+			$response_body = $this->api_client->get_payment_methods();
+
+			if ( is_wp_error( $response_body ) ) {
+				$this->logger->log( $response_body, __FUNCTION__ );
+				return;
+			}
+
+			$payment_methods = $this->get_payment_methods_from_response_body( $response_body );
+			if ( is_wp_error( $payment_methods ) ) {
+				$this->logger->log( $payment_methods, __FUNCTION__ );
+				return;
+			}
+
+			$this->logger->log( 'Successfully loaded payment methods from server response.', __FUNCTION__ );
+			$this->update_last_fetch_timestamp();
+
+			// If we made it this far, it is safe to store the object
+			$this->update_payment_methods( $payment_methods );
+
+			// Potentially auto-update the selected method
```

Comment:
If you're going to put an entire sentence for a method's name, this kind of comment is redundant